python/perspective/setup.py

```diff
@@ -49,14 +50,16 @@ def which(x):
     "future>=0.16.0",
     "numpy>=1.13.1",
     "pandas>=0.22.0",
-    "pyarrow>=0.16.0,<1",
     "python-dateutil>=2.8.0",
     "six>=1.11.0",
     "traitlets>=4.3.2",
 ]
 
 if sys.version_info.major < 3:
     requires.append("backports.shutil-which")
+    requires.append("pyarrow<2")
+else:
+    requires.append("pyarrow>=1.0.1,<2")
```

Comment:
this should match pyproject.toml

Reply:
This version is conditional on python version - does this not make the looser bounds `"pyarrow<2"` for `pyproject.toml` correct?  Perspective under this PR will compile with either `pyarrow`.

Reply:
`pyproject.toml` specifies it as `pyarrow<2` regardless of python version. So it could satisfy with `pyarrow<0.16` even though that won't work. 

You have a couple of options, something like:
`"pyarrow>0.16.0,<2; python_version < '3'", "pyarrow>1.0.1,<2; python_version >= '3'"` would work ok.

Stylistically, every place where we specify requirements should match exactly the behavior of pyproject.toml, otherwise we can very easily start to break our builds and wheels like the state we were in before.